### PR TITLE
Added valSet testcases

### PR DESCRIPTION
**Pull request by 0xsharma**
Added Test-Cases for the validator set mechanisms. This tests the consistency of the updating mechanism of the power and priority of the validators in the validator set.

Link to [Jira Task](https://polygon.atlassian.net/browse/POS-615?atlOrigin=eyJpIjoiMzAyYjI5ODNmNTBhNDgzMmExZGQ5YzFmNWZhODBkN2MiLCJwIjoiaiJ9).

### CODE REVIEW

File: consensus/bor/valset/validator_set_test.go [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+import (
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/crypto"
+	"gotest.tools/assert"
+)
+
+var (
```
Comment (JekaMas) [inline consensus/bor/valset/validator_set_test.go:11]: Could we make it as a function or somehow remove copy-pasting?

File: consensus/bor/valset/validator_set_test.go [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+	//addr4 = 0x168f220B3b313D456eD4797520eFdFA9c57E6C45
+	signer4 = "c8deb0bea5c41afe8e37b4d1bd84e31adff11b09c8c96ff4b605003cce067cd6"
+	key4, _ = crypto.HexToECDSA(signer4)
+	addr4   = crypto.PubkeyToAddress(key4.PublicKey)
+	val4    = NewValidator(addr4, 400)
+)
+
+func TestIncrementProposerPriority(t *testing.T) {
+
+	validators := []*Validator{val1, val2, val3, val4}
```
[the remark this answers is not in the document]
Comment (JekaMas) [reply consensus/bor/valset/validator_set_test.go:39]: The same for all the tests

File: consensus/bor/valset/validator_set_test.go [hunk 3]
```diff
@@ -0,0 +1,102 @@
...
+
+func TestGetValidatorByAddressAndIndex(t *testing.T) {
+	validators := []*Validator{val1, val2, val3, val4}
+	valSet := NewValidatorSet(validators)
+
+	for _, val := range valSet.Validators {
+		idx, valByAddress := valSet.GetByAddress(val.Address)
+		addr, valByIndex := valSet.GetByIndex(idx)
+
+		assert.Equal(t, val.String(), valByIndex.String())
```
Comment (JekaMas) [inline consensus/bor/valset/validator_set_test.go:75]: Could we compare addresses without calling String()?

**Review by JekaMas — COMMENTED** [review]
Looks good to me. Just few comments.

File: consensus/bor/valset/validator_set_test.go [hunk 4]
```diff
@@ -0,0 +1,102 @@
...
+
+	valSet.UpdateWithChangeSet([]*Validator{val3, val4})
+
+	// check totalVotingPower after updating validator set
+	assert.Equal(t, int64(1250), valSet.TotalVotingPower())
+
+	_, updatedVal3 := valSet.GetByAddress(val3.Address)
+	assert.Equal(t, int64(150), updatedVal3.VotingPower)
+
+	_, updatedVal4 := valSet.GetByAddress(val4.Address)
```
Comment (JekaMas) [inline consensus/bor/valset/validator_set_test.go:100]: Could we add negative cases like addresses, indices not from the list?

File: consensus/bor/valset/validator_set_test.go (hunk 1 above)
Comment (0xsharma) [reply consensus/bor/valset/validator_set_test.go:11]: Sure. Will do that !

File: consensus/bor/valset/validator_set_test.go (hunk 3 above)
Comment (0xsharma) [reply consensus/bor/valset/validator_set_test.go:75]: val is basically the pointers to objects containing address, power, priority. As we cannot compare the objects directly. I chose String() as I also wanted to include the power and priority in the comparison and not just the address.

File: consensus/bor/valset/validator_set_test.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (JekaMas) [reply consensus/bor/valset/validator_set_test.go:36]: Let's count test cases that'd be nice to have:
1. number of validators [1, 2, 3, 4]
2. chain length [0, 1, 2, 3, 10]

It's 20 test cases: [Validators, chain] -  [1, 0]; [1, 1], [1, 2], [1, 3], [1, 10], [2, 0], ... [4, 10]

**Review by JekaMas — APPROVED** [review]

File: consensus/bor/valset/validator_set_test.go [hunk 5]
```diff
@@ -0,0 +1,194 @@
...
+	"github.com/ethereum/go-ethereum/crypto"
+	"gotest.tools/assert"
+)
+
+func NewValidatorFromKey(key string, votingPower int64) *Validator {
+	privKey, _ := crypto.HexToECDSA(key)
+	return NewValidator(crypto.PubkeyToAddress(privKey.PublicKey), votingPower)
+}
+
+var (
```
Comment (JekaMas) [inline consensus/bor/valset/validator_set_test.go:16]:
```
func GetValidators() [4]*Validator {
	const (
		// addr1 = 0x96C42C56fdb78294F96B0cFa33c92bed7D75F96a
		signer1 = "c8deb0bea5c41afe8e37b4d1bd84e31adff11b09c8c96ff4b605003cce067cd9"

		// addr2 = 0x98925BE497f6dFF6A5a33dDA8B5933cA35262d69
		signer2 = "c8deb0bea5c41afe8e37b4d1bd84e31adff11b09c8c96ff4b605003cce067cd8"

		//addr3 = 0x648Cf2A5b119E2c04061021834F8f75735B1D36b
		signer3 = "c8deb0bea5c41afe8e37b4d1bd84e31adff11b09c8c96ff4b605003cce067cd7"

		//addr4 = 0x168f220B3b313D456eD4797520eFdFA9c57E6C45
		signer4 = "c8deb0bea5c41afe8e37b4d1bd84e31adff11b09c8c96ff4b605003cce067cd6"
	)

	return [4]*Validator{
		NewValidatorFromKey(signer1, 100),
		NewValidatorFromKey(signer2, 200),
		NewValidatorFromKey(signer3, 300),
		NewValidatorFromKey(signer4, 400),
	}
}
```

Comment (0xsharma) [reply consensus/bor/valset/validator_set_test.go:16]: Sure, will do it this way.

**Review by temaniarpit27 — APPROVED** [review]

**Pull request merged**